File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;
vector<string> p;
bool isPrime(long long n){
    if(n==1) return false;
    for(long long i=2; i*i<=n; i++){
        if(n%i==0) return false;
    }
    return true;
}
int solution(int n, int k) {
  
    //k진수 만들기
    string knum="";
    while(n>0){
        knum = to_string(n%k) + knum;
        n/=k;
    }
    knum += '0';
    //0기준 쪼개기
    long long cur=0;
    int answer=0;
    for(char c:knum){
        if(c=='0'){
            if(cur>1 && isPrime(cur)) answer++;
            cur=0;
        }else{
            cur = cur*10 + (c-'0');
        }
    }
    return answer;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        String knum="";
        while(n>0){
            knum=Integer.toString(n%k)+knum;
            n/=k;
        }
        int answer = 0;
        String[] arr = knum.split("0");     
        for(String s: arr){
            if(s.length()==0) continue;
            //System.out.println(s);
            if(isPrime(Long.parseLong(s))) answer++;
        }
        return answer;
    }
    private boolean isPrime(long num){
        if(num==1) return false;
        if(num==2) return true;
        for(long i=2; i*i<=num ; i++){
            if(num%i==0) return false;
        }
        return true;
    }
}